p2p/peer.py

```diff
@@ -325,9 +330,7 @@ def handle_p2p_msg(self, cmd: protocol.Command, msg: protocol._DecodedMsgType) -
         """Handle the base protocol (P2P) messages."""
         if isinstance(cmd, Disconnect):
             msg = cast(Dict[str, Any], msg)
-            self.logger.debug(
-                "%s disconnected; reason given: %s", self, msg['reason_name'])
-            self.close()
```

Comment:
So this now happens implicitly?

Reply:
Oops, no, we should still `close()` before returning from `read_loop()`, although that's not a big deal as once the read_loop() returns the peer's `run()` method returns as well and everything (including the transport) should get garbage collected.